File: capture-sdk/sdk/src/main/java/net/gini/android/capture/noresults/NoResultsFragmentListener.java
package net.gini.android.capture.noresults;

/**
 * <p>
 * Interface used by {@link NoResultsFragmentCompat} to dispatch events to the hosting Activity.
 * </p>
 */
public interface NoResultsFragmentListener {

    /**
     * <p>
     *     Called when the button on the bottom of the No Results Screen was pressed. This button
     *     should lead the user back to the Camera Screen.
     * </p>
     */
    void onBackToCameraPressed();
}

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/noresults/NoResultsFragmentListener.kt
package net.gini.android.capture.noresults

import net.gini.android.capture.GiniCaptureError

/**
 * Internal use only.
 *
 * Interface used by the {@link NoResultsFragmentCompat} to dispatch events to the hosting activity.
 */
interface NoResultsFragmentListener {

    /**
     * Called when an error occurred.
     *
     * @param error details about what went wrong
     */
    fun onError(error: GiniCaptureError)
}
